java-checks-test-sources/default/src/main/java/checks/ConstantsShouldBeStaticFinalCheck.java

```diff
@@ -79,6 +79,19 @@ static void someMethod2(Object o) {
   }
 }
 
+class AvoidFPsWhenUsingFieldInInstance_SONARJAVA_4749 {
+  private static long nextID = 1L;
+  private final long localID = nextID++; // Compliant
```

Comment:
I think we should check all 4: (prefix, postix) x (increment, decrement), since they are explicitly listed in the implementation